$OpenBSD: patch-program_include_rcmail_php,v 1.1 2013/05/03 10:23:12 sthen Exp $

Fix list page reset when viewing a message in Larry skin (#1489076)

--- program/include/rcmail.php.orig	Fri May  3 11:01:02 2013
+++ program/include/rcmail.php	Fri May  3 11:01:38 2013
@@ -98,7 +98,10 @@ class rcmail extends rcube
 
     // reset some session parameters when changing task
     if ($this->task != 'utils') {
-      if ($this->session && $_SESSION['task'] != $this->task)
+      // we reset list page when switching to another task
+      // but only to the main task interface - empty action (#1489076)
+      // this will prevent from unintentional page reset on cross-task requests
+      if ($this->session && $_SESSION['task'] != $this->task && empty($this->action))
         $this->session->remove('page');
       // set current task to session
       $_SESSION['task'] = $this->task;
